yt_dlp/extractor/vk.py

```diff
@@ -606,39 +640,109 @@ class VKUserVideosIE(VKBaseIE):
+        access_token = self._get_access_token(page_id)
+        if section.startswith('play'):
+            url, payload = self._get_playlist_payload_and_url(page_id, album_id, access_token)
+        else:
+            url, payload = self._get_user_payload_and_url(url, access_token)
+
+        next_token = None
+        section_id = None
+        data = self._download_json(
+            url,
+            page_id,
+            query=payload,
+            headers={
+                'content-type': 'application/x-www-form-urlencoded',
+                'origin': 'https://vkvideo.ru',
+            },
+        )['response']
+        video_list = traverse_obj(data, ('catalog', 'videos', {list}), ('videos', {list}), ('items', {list}))
+        count = len(video_list)
+        total = traverse_obj(data, ('albums', 0, 'count', {int}), ('count', {int}))
+        next_token = traverse_obj(data, ('catalogsections', 0, 'next_from'), (('catalog', 'sections', 1, 'next_from')))
+        section_id = traverse_obj(data, ('catalog', 'default_section'))
         while True:
             for video in video_list:
-                v = self._VIDEO._make(video[:2])
-                video_id = '%d_%d' % (v.owner_id, v.id)
-                yield self.url_result(
-                    'https://vk.com/video' + video_id, VKIE.ie_key(), video_id)
+                vid = video.get('id')
+                void = video.get('owner_id')
+                video_id = f'{void}_{vid}'
+                yield self._parse_videos(video, video_id)
             if count >= total:
                 break
-            video_list_json = self._download_payload('al_video', page_id, {
-                'act': 'load_videos_silent',
-                'offset': count,
-                'oid': page_id,
-                'section': section,
-            })[0][section]
-            new_count = video_list_json['count']
+            if section.startswith('play'):
+                url, payload = self._get_playlist_payload_and_url(page_id, album_id, access_token, offset=count)
+            else:
+                url, payload = self._get_user_payload_and_url(None, access_token, next_token, section_id=section_id)
+            video_list_json = self._download_json(
+                url,
+                page_id,
+                query=payload,
+                headers={
+                    'content-type': 'application/x-www-form-urlencoded',
+                    'origin': 'https://vkvideo.ru',
+                },
+            )['response']
+            video_list = traverse_obj(video_list_json, ('videos'), ('items',))
+            if not video_list:
+                break
+            new_count = len(video_list)
+            next_token = traverse_obj(video_list_json, ('section', 'next_from'))
             if not new_count:
                 self.to_screen(f'{page_id}: Skipping {total - count} unavailable videos')
                 break
             count += new_count
```

Comment:
The suite of `if not new_count:` is unreachable.
```suggestion
            if not video_list:
                self.to_screen(f'{page_id}: Skipping {remaining} unavailable videos')
                break
            next_token = traverse_obj(video_list_json, ('section', 'next_from'))
```